controller/config.go

```diff
@@ -313,6 +318,107 @@ func LoadConfig(path string) (*Config, error) {
 		panic("controllerConfig must provide [db] or [raft]")
 	}
 
+	//SPIFFE Trust Domain
+	var spiffeId *url.URL
+	if controllerConfig.Raft != nil {
+		//HA setup, SPIFFE ID must come from certs
+		var err error
+		spiffeId, err = GetSpiffeIdFromIdentity(controllerConfig.Id.Identity)
+		if err != nil {
+			panic("error determining a trust domain from a SPIFFE id in the root identity for HA configuration, must have a spiffe:// URI SANs in the server certificate or along the signing CAs chain: " + err.Error())
+		}
+
+		if spiffeId == nil {
+			panic("unable to determine a trust domain from a SPIFFE id in the root identity for HA configuration, must have a spiffe:// URI SANs in the server certificate or along the signing CAs chain")
+		}
+	} else {
+		// Non-HA/legacy system, prefer SPIFFE id from certs, but fall back to configuration if necessary
+		spiffeId, _ = GetSpiffeIdFromIdentity(controllerConfig.Id.Identity)
+
+		if spiffeId == nil {
+			//for non HA setups allow the trust domain to come from the configuration root value `trustDomain`
+			if value, found := cfgmap["trustDomain"]; found {
```

Comment:
it really would be swell to get important constants like "trustDomain" documented and into a constants type location...

Reply:
I had a similar thought; I followed the pattern in the file. I am not going to do it in this PR. If someone actually needs those constants they can go through and do it.